docs/admin/federation/kubefed.md

```diff
@@ -0,0 +1,172 @@
+
+```shell
+curl -O https://storage.googleapis.com/kubernetes-release/release/v1.5.0/kubernetes-client-linux-amd64.tar.gz
+tar -xzvf kubernetes-client-linux-amd64.tar.gz
+sudo cp kubernetes/client/bin/kubefed /usr/local/bin
+sudo chmod +x /usr/local/bin/kubefed
+sudo cp kubernetes/client/bin/kubectl /usr/local/bin
+sudo chmod +x /usr/local/bin/kubectl
+```
+
+## Choosing a host cluster.
+
+Choose one of your Kubernetes clusters as a cluster where you want to
+host the federation control plane components. We refer to this cluster
+as a "host cluster" henceforth. Ensure that you have a kubeconfig entry
+in your local kubeconfig corresponding to this cluster. You can verify
+this by running:
+
+```shell
+kubectl config get-contexts
+```
+
+and ensuring that there is a context corresponding to your host
+cluster.
+
+Please also make a note of this context because you will need it next.
+
+
+## Deploying a federation control plane.
+
+Deploying a federation control plane on your host cluster is as simple
+as running:
+
+```shell
+kubefed init fellowship --host-cluster-context=rivendell  --dns-zone-name="example.com"
+```
+
+where `fellowship` is the name of the federation, `rivendell` is the
+host cluster context from the previous step and `example.com` is the
+domain name suffix that you want for your federated services. This has
+to be a real domain that you control over and is programmable by your
+DNS provider.
+
+`kubefed init` sets up the federation control plane in the host
+cluster and also adds an entry for the federation API server in your
+local kubeconfig. However, in this alpha release it does not
+automatically set the current context to the newly deployed federation.
+You can set this by running:
+
+```shell
+kubectl config use-context fellowship
+```
+
+where `fellowship` is the name of your federation.
+
+## Joining a cluster
+
+Cluster Federation allows you to manage your workload across multiple
+clusters. In order to do that, a federation control plane must be
+aware of the clusters that it is responsible for managing. You can
+teach the federation control plane about a cluster by joining that
+cluster to the federation.
+
+To join a cluster to a federation run:
+
+```
+kubfed join gondor --host-cluster-context=rivendell
+```
+
+where `gondor` is the name of the cluster you are joining and
+`rivendell` is the federation control plane's host cluster.
+
+### Naming rules and customization
+
+#### Cluster name
+
+Cluster name supplied to `kubefed join` must be a valid RFC 1035 label.
+
+Furthermore, federation control plane requires credentials of the
+joined clusters to operate on them. These credentials are obtained
+from the local kubeconfig. `kubefed join` uses the cluster name
+specified as the argument to look for the cluster's context in the
+local kubeconfig. If it fails to find a matching context, it exits
+with an error.
+
+It is particularly a problem in the cases where context names
```

Comment:
"This might cause issues in cases where context names for each cluster in the federation don't follow RFC 1035 label naming rules. In such cases, you can specify a cluster name that conforms..."